client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java

```diff
@@ -137,10 +136,22 @@ public int getSpaceIdFromCache(String name) {
      * @return
      */
     public List<SpaceNameID> listSpaces() {
+        int retry = 1;
         ListSpacesReq request = new ListSpacesReq();
-        ListSpacesResp response;
+        ListSpacesResp response = null;
         try {
-            response = client.listSpaces(request);
+            while (retry-- >= 0) {
+                response = client.listSpaces(request);
+                if (response.code == ErrorCode.E_LEADER_CHANGED) {
+                    close();
+                    HostAddr newLeader = response.getLeader();
```

Comment:
I think here is better to provide a function to return a new `metaClient`, and it's not only happening in `listSpaces`, other meta interfaces also do it too.

Reply:
Great idea. Leader change will happen  at any time.